src/Pact/Types/Command.hs

```diff
@@ -258,35 +258,32 @@ instance FromJSON UserSig where
     UserSig <$> o .: "sig"
 
 
-
-data CommandError = CommandError {
-      _ceMsg :: String
-    , _ceDetail :: Maybe String
-}
-instance ToJSON CommandError where
-    toJSON (CommandError m d) =
-        object $ [ "status" .= ("failure" :: String)
-                 , "error" .= m ] ++
-        maybe [] ((:[]) . ("detail" .=)) d
-
-newtype CommandSuccess a = CommandSuccess { _csData :: a }
+newtype PactResult = PactResult (Either PactError PactValue)
   deriving (Eq, Show)
-
-instance (ToJSON a) => ToJSON (CommandSuccess a) where
-    toJSON (CommandSuccess a) =
-        object [ "status" .= ("success" :: String)
-               , "data" .= a ]
-
-instance (FromJSON a) => FromJSON (CommandSuccess a) where
-    parseJSON = withObject "CommandSuccess" $ \o ->
-        CommandSuccess <$> o .: "data"
-
-data CommandResult = CommandResult
-  { _crReqKey :: RequestKey
-  , _crTxId :: Maybe TxId
-  , _crResult :: Value
-  , _crGas :: Gas
-  } deriving (Eq,Show)
+instance ToJSON PactResult where
+  toJSON (PactResult (Right s)) =
+    object [ "status" .= ("success" :: String)
+           , "data" .= s ]
+  toJSON (PactResult (Left f)) =
+    object [ "status" .= ("failure" :: String)
+           , "data" .= f ]
+instance FromJSON PactResult where
+  parseJSON (A.Object o) = PactResult <$>
+                           ((Left <$> o .: "data") <|>
+                            (Right <$> o .: "data"))
+  parseJSON p = fail $ "Invalid PactResult " ++ show p
+
+data CommandResult l = CommandResult
+  { _crReqKey :: !RequestKey
+  , _crTxId :: !(Maybe TxId)
+  , _crResult :: !PactResult
+  , _crGas :: !Gas
+  , _crLogs :: !(Maybe l)                 -- Level of logging (i.e. full TxLog vs hashed logs)
```

Comment:
go ahead and add haddock docs to each field

Reply:
Added the following haddock: https://github.com/kadena-io/pact/pull/504/commits/1af4600fa58cdba9e9c3114fa517102791ec8f6b#diff-07f1972f68441e4bcfba09fc7666f694R277

Reply:
that's not haddock format, see https://github.com/kadena-io/pact/blob/ba34cf9409db4705cf6b8e5ff35f3c6b0a6e7766/src/Pact/Types/Runtime.hs#L194

Reply:
ie, give a top-level comment `-- | API result for Pact execution`, and then put `-- | foo bar` before each field line